ruoyi-common/src/test/java/com/ruoyi/common/support/ConvertTest.java

```diff
@@ -0,0 +1,205 @@
+        assertEquals((byte) 8, (byte) toByte("foo", (byte) 8));
+        assertEquals((byte) 1, (byte) toByte(1));
+    }
+
+    @Test
+    public void testToShort() {
+        assertEquals((short) 8, (short) toShort(null, (short) 8));
+        assertEquals((short) 1, (short) toShort((short) 1, (short) 8));
+        assertEquals((short) 8, (short) toShort("", (short) 8));
+        assertEquals((short) 123, (short) toShort("123", (short) 8));
+        assertEquals((short) 8, (short) toShort("foo", (short) 8));
+        assertEquals((short) 1, (short) toShort(1));
+    }
+
+    @Test
+    public void testToNumber() {
+        assertEquals(8, toNumber(null, 8));
+        assertEquals(8, toNumber("", 8));
+        assertEquals((long) 123, toNumber("123", 8));
+        assertEquals(8, toNumber("foo", 8));
+        assertEquals(1, toNumber(1));
+    }
+
+    @Test
+    public void testToInt() throws ParseException {
+        assertEquals(8, (int) toInt(null, 8));
+        assertEquals(1, (int) toInt(NumberFormat.getInstance().parse("1"), 8));
+        assertEquals(8, (int) toInt("", 8));
+        assertEquals(123, (int) toInt("123", 8));
+        assertEquals(8, (int) toInt("foo", 8));
+        assertEquals(1, (int) toInt(1));
+    }
+
+    @Test
+    public void testToIntArray() {
+        assertEquals(Arrays.asList(), Arrays.asList(toIntArray(null, "")));
+        assertEquals(Arrays.asList(0, 0, 0, 0), Arrays.asList(toIntArray("a,b,c,d")));
+    }
+
+    @Test
+    public void testToLongArray() {
+        assertEquals(Arrays.asList(), Arrays.asList(toLongArray(null, "")));
+        assertEquals(Arrays.asList(null, null, null, null), Arrays.asList(toLongArray("a,b,c,d")));
+    }
+
+    @Test
+    public void testToStrArray() {
+        assertEquals(Arrays.asList("a", "b", "c", "d"), Arrays.asList(toStrArray("a,b,c,d")));
+    }
+
+    @Test
+    public void testToLong() {
+        assertEquals(8, (long) toLong(null, (long) 8));
+        assertEquals(1, (long) toLong((long) 1, (long) 8));
+        assertEquals(8, (long) toLong("", (long) 8));
+        assertEquals(123, (long) toLong("123", (long) 8));
+        assertEquals(8, (long) toLong("foo", (long) 8));
+        assertEquals(1, (long) toLong(1));
+    }
+
+    @Test
+    public void testToDouble() {
+        assertEquals(8, toDouble(null, (double) 8), 0);
+        assertEquals(1, toDouble((double) 1, (double) 8), 0);
+        assertEquals(8, toDouble("", (double) 8), 0);
+        assertEquals(123, toDouble("123", (double) 8), 0);
+        assertEquals(8, toDouble("foo", (double) 8), 0);
+        assertEquals(1, toDouble(1), 0);
+    }
+
+    @Test
+    public void testToFloat() {
+        assertEquals(8, toFloat(null, (float) 8), 0);
+        assertEquals(1, toFloat((float) 1, (float) 8), 0);
+        assertEquals(8, toFloat("", (float) 8), 0);
+        assertEquals(123, toFloat("123", (float) 8), 0);
+        assertEquals(8, toFloat("foo", (float) 8), 0);
```

Comment:
Instead of casting to `(float)` in the second param of `toFloat`, you could use a double literal (`8F`)

Reply:
Done